wire-runtime/src/main/java/com/squareup/wire/Message.java

```diff
@@ -149,12 +147,30 @@ protected Builder() {
      * this message has no unknown fields.
      */
     public final ByteString buildUnknownFields() {
-      return unknownFieldsBuffer != null
-          ? unknownFieldsBuffer.clone().readByteString()
-          : ByteString.EMPTY;
+      if (unknownFieldsBuffer != null) {
+        // Reads and caches the unknown fields from the buffer.
+        unknownFieldsByteString = unknownFieldsBuffer.readByteString();
+        unknownFieldsBuffer = null;
```

Comment:
Why are you clearing `unknownFieldsBuffer` and `unknownFieldsBuffer`? What if the caller adds an additional field and calls this method a second time?